bentoml/_internal/io_descriptors/json.py

```diff
@@ -165,25 +156,26 @@ def openapi_responses_schema(self) -> t.Dict[str, t.Any]:
 
     async def from_http_request(self, request: Request) -> JSONType:
         json_str = await request.body()
-        if self._pydantic_model is not None and self._validate_json:
-            try:
-                import pydantic
-            except ImportError:
-                raise MissingDependencyException(
-                    "`pydantic` must be installed to use `pydantic_model`"
-                ) from None
+        try:
+            json_obj = json.loads(json_str)
+        except json.JSONDecodeError as e:
+            raise BadInput(f"Invalid JSON input received: {e}") from None
+
+        if self._pydantic_model is not None:
             try:
-                pydantic_model = self._pydantic_model.parse_raw(json_str)
+                pydantic_model = self._pydantic_model.parse_obj(json_obj)
                 return pydantic_model
             except pydantic.ValidationError as e:
-                raise BadInput(f"Json validation error: {e}") from None
+                raise BadInput(f"Invalid JSON input received: {e}") from None
         else:
-            try:
-                return json.loads(json_str)
-            except json.JSONDecodeError as e:
-                raise BadInput(f"Json validation error: {e}") from None
+            return json_obj
+
+    async def to_http_response(
+        self, obj: JSONType | pydantic.BaseModel, ctx: Context | None = None
+    ):
+        if isinstance(obj, pydantic.BaseModel):
+            obj = obj.dict()
```

Comment:
this may not be used so commonly, but it allows user to use a pydantic instance as API return value